kafka/consumer.c

```diff
@@ -43,9 +44,28 @@ consumer_poll_loop(void *arg) {
             rd_msg = rd_kafka_consumer_poll(poller->rd_consumer, 1000);
             if (rd_msg != NULL) {
                 if (rd_msg->err != RD_KAFKA_RESP_ERR_NO_ERROR) {
-                    // FIXME: push errors to error queue?
+                    errors_count++;
+
+                    error_msg_t *msg = new_error_msg(rd_msg->err, rd_kafka_err2str(rd_msg->err));
+
+                    // free rdkafka message instantly to prevent hang on close / destroy consumer
                     rd_kafka_message_destroy(rd_msg);
+                    rd_msg = NULL;
+
+                    pthread_mutex_lock(&event_queues->queues[ERROR_QUEUE]->lock);
```

Comment:
Also you need to check that event_queues->queues[ERROR_QUEUE] actually exists.

Reply:
queue_push make same check inside, but we can check, that event_queues->queues exists